ComponentKit/Core/Scope/CKComponentScopeFrame.mm

```diff
@@ -57,27 +57,47 @@ + (CKComponentScopeFramePair)childPairForPair:(const CKComponentScopeFramePair &
   CKCAssert([componentClass isSubclassOfClass:[CKComponent class]], @"%@ is not a component", NSStringFromClass(componentClass));
   CKAssertNotNil(pair.frame, @"Must have frame");
 
-  // Find the existing child, if any, in the old frame
-  CKComponentScopeFrame *existingChild;
+  CKComponentScopeFrame *existingChildFrameOfEquivalentPreviousFrame;
   if (pair.equivalentPreviousFrame) {
     const auto &equivalentPreviousFrameChildren = pair.equivalentPreviousFrame->_children;
     const auto it = equivalentPreviousFrameChildren.find({componentClass, identifier});
-    existingChild = (it == equivalentPreviousFrameChildren.end()) ? nil : it->second;
+    existingChildFrameOfEquivalentPreviousFrame = (it == equivalentPreviousFrameChildren.end()) ? nil : it->second;
+  }
+
+  const auto existingChild = pair.frame->_children.find({componentClass, identifier});
+  if (!pair.frame->_children.empty() && (existingChild != pair.frame->_children.end())) {
+    /*
+     The component was involved in a scope collision and the scope handle needs to be reacquired.
+     In the event of a component scope collision the component scope frames reuses the existing scope handle; any
+     existing state will be made available to the component that introduced the scope collision. This leads to some
+     interesting side effects:
+
+       1. Any component state associated with the scope handle will be shared between components with colliding scopes
+       2. Any component controller associated with the scope handle will be responsible for each component with
+          colliding scopes; resulting in strange behavior while components are mounted, unmounted, etc.
+
+     Reusing the existing scope handle allows ComponentKit to detect component scope collisions during layout. Moving
```

Comment:
Hm. Technically we're reusing the same handle, we just create a new instance with the exact same state so it can be reacquired. An entirely new frame is created to hold that handle.